### ISSUE TITLE
pusher_1  No admin backoffice set!

### ISSUE THREAD

**Issue by X52p** [thecodingmachine/workadventure#846]
When using the default Docker compose file that packaged with the release (1.1), I get an 500 as a reply to a get request to the pusher.

The Get request:
`GET <pusher.url>map?organizationSlug=tcm&worldSlug=workadventure&roomSlug=lyon`

The Server Log shows this error over and over again:
`pusher_1         | An error happened No admin backoffice set!`

**Comment by Findus23** [comment]
I also can't get any further than this using the default development docker-compose.yaml.

It seems like this is caused by the fact that the `ADMIN_API_URL` environment variable is not set:

https://github.com/thecodingmachine/workadventure/blob/3d1c2dc05a679f23ae6d6f481731c4d9ea6db1dc/pusher/src/Services/AdminApi.ts#L48-L50

Unfortunately there are no information anywhere on how it should look like. The only places it is mentioned is 
https://github.com/thecodingmachine/workadventure/blob/94333a74389406eb4bc6acba0d3620ca184ea647/deeployer.libsonnet#L26
and
https://github.com/thecodingmachine/workadventure/blob/a1f06ffbd58390207772f45e5f31c7234f1f821d/contrib/docker/docker-compose.prod.yaml#L88

but both of them don't mention what it should be set to.

I guessed `ADMIN_URL: //api.workadventure.localhost`, but that just brings a "connection refused" as api.workadventure.localhost doesn't seem to work (I get a `Bad Gateway`, but nothing in the docker logs).

So I give up for now until someone else has another idea.

**Comment by miterion** [comment]
We still have no idea how to avoid this error but you can use workadventure without setting the `ADMIN_API_URL`. The problem @X52p describes was a symptom of a misconfiguration in our apache reverse proxy and had nothing to do with the admin backend.

So tl;dr If this is the only error you see, you might have some other problem somewhere else.

**Comment by Findus23** [comment]
@miterion You are right. On a closer look it really seems like the `ADMIN_API_URL` has nothing to do with why my instance doesn't work. 

But I'm still wondering what I do wrong. I didn't modify any file and everything works up until the request to `http://pusher.workadventure.localhost/map?organizationSlug=tcm&worldSlug=workadventure&roomSlug=lyon` which returns a 500 and `An error occurred`, but doesn't print anything on the error log (apart from this error here).

**Comment by Findus23** [comment]
Okay, this is going off-topic here, but I resolved my issue and want to document it here for the next person to stumble across it (as it is really stupid).

It turns out that the failing  `pusher.workadventure.localhost/map` request was also a red hering and is working "as intended" (as it only forwards the request to the admin api which doesn't exist here).

But directly next to it, the request to the websocket was also failing and it turns out that I enabled HTTPS-only-mode in Firefox, but was using the non-HTTPS localhost site for testing (which worked fine as I guess localhost is an exception. But for whatever reason Firefox doesn't allow non-HTTPS websocket requests to localhost domains in HTTPS-only-mode.
